frontend/src/metabase/components/SingleSelectListField/SingleSelectListField.styled.tsx

```diff
@@ -39,21 +40,23 @@ export const OptionItem = styled.div<OptionItemProps>`
   margin: 0;
   padding: 0.5rem 0.6rem;
   width: 100%;
-  background-color: ${props =>
-    props.selected ? props.selectedColor : "var(--mb-color-background)"};
-  color: ${props =>
+  ${props =>
     props.selected
-      ? "var(--mb-color-text-white)"
-      : "var(--mb-color-text-primary)"};
+      ? css`
```

Comment:
2 things here:
1. group styles with `css` instead of separating variants by property values. This should make it easier to understand.
2. Update selected/hover styles.